### PR TITLE
#1908: Disable preConfig

### LINKED ISSUE #1908: Disable debug print preConfig evaulation when in production mode
**What Needs to be Done?**

- [ ] First do this
- [ ] Then do that.
- [ ] Also this other thing.

**Is your feature request related to a problem? Please describe.**

**Describe potential solution outcome**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by stmcgovern**
Closes #1908.

### CODE REVIEW

**Comment by stmcgovern** [comment]
From grouped PR, see:
https://github.com/DARMA-tasking/vt/pull/1916#discussion_r952919356

**Review by PhilMiller — CHANGES_REQUESTED** [review]
Could you pick up my suggestion from the earlier combined PR?

File: src/vt/configs/debug/debug_print.h [hunk 1]
```diff
@@ -83,6 +83,10 @@
 #define vt_debug_argument_option(opt)                                   \
   ::vt::debug::preConfig()->vt_debug_ ## opt
 
+#if vt_check_enabled(production_build)
+#define vt_debug_print_impl(force, inconfig, inmode, cat, ctx, ...)     \
+  do { if (false) vt_force_use(__VA_ARGS__); } while (false);
```
Comment (PhilMiller) [inline src/vt/configs/debug/debug_print.h:88]: The point of using do-while in a macro is that it forces the caller to treat it like a full statement, with a following semicolon. That only works if the macro definition leaves it off

Comment (stmcgovern) [reply src/vt/configs/debug/debug_print.h]: Yes - fixed.

**Review by PhilMiller — APPROVED** [review]

**Pull request merged**